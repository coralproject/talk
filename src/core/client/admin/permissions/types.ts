import { GQLUSER_ROLE_RL } from "coral-framework/schema";

export type PermissionMap<AbilityType extends string, ContextType> = {
  [abilityType in AbilityType]: {
    [role in GQLUSER_ROLE_RL]?: (context?: ContextType) => boolean;
  };
};
