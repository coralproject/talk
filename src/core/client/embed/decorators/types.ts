import pym from "pym.js";

export type CleanupCallback = () => void;
export type Decorator = (pym: pym.Parent) => CleanupCallback | void;
