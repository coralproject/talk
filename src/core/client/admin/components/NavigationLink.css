.root {
  color: var(--palette-text-secondary);
  font-family: var(--font-family-sans-serif);
  font-weight: var(--font-weight-bold);
  font-size: calc(18rem / var(--rem-base));
  line-height: calc(20em / 18);
  letter-spacing: calc(-0.1em / 18);
  height: 100%;
  display: inline-flex;
  align-items: center;
  padding: 0 calc(1.5 * var(--spacing-unit));
  text-transform: uppercase;
  text-decoration: none;
  &:hover {
    cursor: pointer;
  }
}

.active {
  background-color: var(--palette-brand);
  text-decoration: none;
  color: var(--palette-text-light);
}
