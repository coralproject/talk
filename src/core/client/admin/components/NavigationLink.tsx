import { Link, LocationDescriptor } from "found";
import React, { StatelessComponent } from "react";

import styles from "./NavigationLink.css";

interface Props {
  children: React.ReactNode;
  to: string | LocationDescriptor;
}

const NavigationLink: StatelessComponent<Props> = props => (
  <Link to={props.to} className={styles.root} activeClassName={styles.active}>
    {props.children}
  </Link>
);

export default NavigationLink;
