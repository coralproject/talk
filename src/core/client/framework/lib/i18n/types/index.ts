export { default as FluentShortNumber } from "./FluentShortNumber";
