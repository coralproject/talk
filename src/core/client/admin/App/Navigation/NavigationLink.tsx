import { Link, LocationDescriptor } from "found";
import React, { FunctionComponent } from "react";

import { AppBarNavigationItem } from "coral-admin/ui/components";

interface Props {
  children: React.ReactNode;
  to: string | LocationDescriptor;
}

const NavigationLink: FunctionComponent<Props> = props => (
  <Link to={props.to} as={AppBarNavigationItem} activePropName="active">
    {props.children}
  </Link>
);

export default NavigationLink;
