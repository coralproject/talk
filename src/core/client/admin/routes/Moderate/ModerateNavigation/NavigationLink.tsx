import { Link, LocationDescriptor } from "found";
import React, { FunctionComponent } from "react";

import { SubBarNavigationItem } from "coral-ui/components/v2";

interface Props {
  children: React.ReactNode;
  to: string | LocationDescriptor;
}

const NavigationLink: FunctionComponent<Props> = props => (
  <Link to={props.to} as={SubBarNavigationItem} activePropName="active">
    {props.children}
  </Link>
);

export default NavigationLink;
