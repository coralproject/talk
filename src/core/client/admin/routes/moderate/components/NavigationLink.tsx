import { Link, LocationDescriptor } from "found";
import React, { StatelessComponent } from "react";

import { SubBarNavigationItem } from "talk-ui/components";

interface Props {
  children: React.ReactNode;
  to: string | LocationDescriptor;
}

const NavigationLink: StatelessComponent<Props> = props => (
  <Link to={props.to} Component={SubBarNavigationItem} activePropName="active">
    {props.children}
  </Link>
);

export default NavigationLink;
