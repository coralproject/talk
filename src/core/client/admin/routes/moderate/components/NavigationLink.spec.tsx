import React from "react";
import { createRenderer } from "react-test-renderer/shallow";

import { PropTypesOf } from "talk-framework/types";

import NavigationLink from "./NavigationLink";

it("renders correctly", () => {
  const props: PropTypesOf<typeof NavigationLink> = {
    to: "/moderate",
    children: "link",
  };
  const renderer = createRenderer();
  renderer.render(<NavigationLink {...props} />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
