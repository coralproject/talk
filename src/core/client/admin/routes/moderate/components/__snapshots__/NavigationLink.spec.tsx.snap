// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ForwardRef(render)
  activePropName="active"
  as={[Function]}
  to="/moderate"
>
  link
</ForwardRef(render)>
`;
